app/src/features/FhirResourceTree/resourceTreeUtils.ts

```diff
@@ -231,13 +229,24 @@ export const buildTree = (
   if (isElementNodeChildOf(currentElementNode, previousElementNode)) {
     if (
       previousElementNode.isArray &&
-      previousElementNode.type === "BackboneElement"
+      previousElementNode.type === "BackboneElement" &&
+      !currentElementNode.sliceName
     ) {
       previousElementNode.backboneElementDefinitions.push(
         currentElementDefinition
       );
       buildTree(rest, rootNode, previousElementNode);
     } else {
+      if (currentElementNode.sliceName) {
```

Comment:
could you declare a function for this (or split into multiple functions) to describe more what's going on ?

Reply:
I am currently reworking it. I'm kind of struggling with it...